chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -417,10 +417,52 @@ def self.size_to_bytes(s)
       end
     end
 
+    class DependsOn
+      class << self
+        def add(dependency)
+          @recipe_depedencies ||= Mash.new
+          @recipe_depedencies.merge!(dependency)
+        end
+
+        def get(recipe)
+          @recipe_depedencies.fetch(recipe, [])
+        end
+      end
+    end
+
     class Config
       class << self
         attr_accessor :node
 
+        def last_two_configs(group, barclamp, instance)
+          prev_cfg = load(group, barclamp, instance)
+          cur_cfg = node[barclamp.to_sym].current_default
```

Comment:
I had to choose/use `current_default` attribute here to get the underlying hash, however this includes the default keys, since some of these are never saved to databag, they do not show up in `prev_cfg` but always in `cur_cfg`. 

can someone suggest and way to not have these used for `prev_cfg, cur_cfg` comparison  ?